### ISSUE TITLE
Add version field and custom date in the model and the Report

### ISSUE THREAD

**Issue by KANDRIANO** [OWASP/threat-dragon#1621]
**Describe what problem your feature request solves**:  

When I export the model to PDF, it only shows the date generated.
Showing the version number of the model (eg 0.1, 1.0, 2.0) and a custom date would be preferable.

**Describe the solution you'd like**:  
2 new fields "Version" and "Date" are needed in the edit page.
The Version and custom Date value must then be exported in the PDF, either in place of date generated or maybe as a subtitle below the title, with smaller font. (an option to select placement either under the contributors or under the title would be a nice to have, but not necessary)

**Declaration**:  

By submitting this issue you have:  

- [x] read the [contribution guide](../blob/main/contributing.md) and agree to the [Code of Conduct](../blob/main/code_of_conduct.md)
- [x] not used agentic or generative AI in creating this feature request. The current feature request was written 100% manually without the use of LLM or anything genAI related.

**Additional context**:  

<img width="2475" height="856" alt="Image" src="https://github.com/user-attachments/assets/4b263686-b8a8-4c61-bd30-3c783654a3ec" />

**Comment by jgadsden** [comment]
This is a great first issue for someone! If you are new to Threat Dragon, reading the docs and understanding the code is a great first step! You are also uniquely qualified to find blind-spots in the documentation as you fix this. 😎

_**This should not be done using Generative AI or an agent**_. "Good first issues" are meant for our human friends to learn and start participating. If you have questions after being assigned the issue, I'm happy to help! Again, please read and understand the [contribution guide](https://github.com/OWASP/threat-dragon/blob/main/contributing.md) before starting to work on this issue. You should be assigned before opening a pull request.

**Comment by praxzy** [comment]
Hi, I would like to work on this issue.

**Comment by jgadsden** [comment]
OK, thanks @praxzy 
there needs to be a discussion on what the date field should contain - published date or generated date?
I suggest that if there is no published date entered then the generated date should take its place
both of these new values should be stored as key:value pairs in the threat model itself

Suggest two new keys `released_at` and `release_version` to fit in with TM-BOM-like schemas

And note that AI is expressly forbidden for this issue
